File: app/src/main/java/com/waxym/defibapp/RecyclerDefibAdapter.java
package com.waxym.defibapp;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

public class RecyclerDefibAdapter extends RecyclerView.Adapter<RecyclerDefibAdapter.ViewHolder> {

    private List<Record> list;
    private LayoutInflater mInflater;
    private ItemClickListener mClickListener;
    private Context context;

    RecyclerDefibAdapter(Context context, List<Record> list) {
        this.context = context;
        this.mInflater = LayoutInflater.from(context);
        this.list = list;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = mInflater.inflate(R.layout.defib_item, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        Record currentRecord = list.get(i);
        viewHolder.textViewLocationName.setText(String.format(context.getString(R.string.site_name), currentRecord.getFields().getNomSite()));
        viewHolder.textViewAdress.setText(String.format(context.getString(R.string.adress), currentRecord.getFields().getAdresse()));
        viewHolder.textViewDistance.setText(String.format(context.getString(R.string.meter), String.valueOf(currentRecord.getFields().getDistance())));
        if(currentRecord.getFields().getImplantation() != null){
            viewHolder.textViewImplantation.setText(String.format(context.getString(R.string.implantation), currentRecord.getFields().getImplantation()));
        } else {
            viewHolder.textViewImplantation.setText(context.getString(R.string.no_implantation));
        }
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public Record getItem(int id) {
        return list.get(id);
    }

    public void setClickListener(ItemClickListener itemClickListener) {
        this.mClickListener = itemClickListener;
    }

    public interface ItemClickListener {
        void onItemClick(View view, int position);
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        TextView textViewLocationName;
        TextView textViewAdress;
        TextView textViewDistance;
        TextView textViewImplantation;

        ViewHolder(View itemView) {
            super(itemView);
            textViewLocationName = itemView.findViewById(R.id.textViewLocationName);
            textViewAdress = itemView.findViewById(R.id.textViewAdress);
            textViewDistance = itemView.findViewById(R.id.textViewDistance);
            textViewImplantation = itemView.findViewById(R.id.textViewImplantation);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            if (mClickListener != null) mClickListener.onItemClick(view, getAdapterPosition());
        }
    }
}

File: app/src/main/java/com/waxym/defibapp/RecyclerDefibAdapter.kt
package com.waxym.defibapp

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

class RecyclerDefibAdapter internal constructor(private val context: Context, private val list: List<Record>) : RecyclerView.Adapter<RecyclerDefibAdapter.ViewHolder>() {
    private val mInflater: LayoutInflater = LayoutInflater.from(context)
    private var mClickListener: ItemClickListener? = null

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val view = mInflater.inflate(R.layout.defib_item, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        val currentRecord = list[i]
        viewHolder.textViewLocationName.text = context.getString(R.string.site_name).plus(" " + currentRecord.fields!!.nomSite)
        viewHolder.textViewAdress.text = context.getString(R.string.adress).plus(" " + currentRecord.fields!!.adresse)
        viewHolder.textViewDistance.text = currentRecord.fields!!.distance.toString().plus(context.getString(R.string.meter))
        if (currentRecord.fields!!.implantation != null) {
            viewHolder.textViewImplantation.text = context.getString(R.string.implantation).plus(" " + currentRecord.fields!!.implantation)
        } else {
            viewHolder.textViewImplantation.text = context.getString(R.string.no_implantation)
        }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    fun getItem(id: Int): Record {
        return list[id]
    }

    fun setClickListener(itemClickListener: ItemClickListener) {
        this.mClickListener = itemClickListener
    }

    interface ItemClickListener {
        fun onItemClick(view: View, position: Int)
    }

    inner class ViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

        internal var textViewLocationName: TextView = itemView.findViewById(R.id.textViewLocationName)
        internal var textViewAdress: TextView = itemView.findViewById(R.id.textViewAdress)
        internal var textViewDistance: TextView = itemView.findViewById(R.id.textViewDistance)
        internal var textViewImplantation: TextView = itemView.findViewById(R.id.textViewImplantation)

        init {
            itemView.setOnClickListener(this)
        }

        override fun onClick(view: View) {
            if (mClickListener != null) mClickListener!!.onItemClick(view, adapterPosition)
        }
    }
}